fashion_mnist/README.md

```diff
@@ -0,0 +1,394 @@
+
+## Modify fashion_mnist.py
+
+Double-click `fashion_mnist.py` in the file picker, which will open it in the editor:
+
+![image](https://user-images.githubusercontent.com/16640218/53517877-8c924100-3a84-11e9-9a65-a9054529cc6a.png)
+
+Let's dive into the modifications!
+
+### Add Horovod import
+
+Add the following code after `import tensorflow as tf`:
+
+```python
+import horovod.keras as hvd
+```
+
+![image](https://user-images.githubusercontent.com/16640218/53517965-c8c5a180-3a84-11e9-9b36-e745bebe84df.png)
+
+### Initialize Horovod
+
+Add the following code after `args = parser.parse_args()`:
+
+```python
+# Horovod: initialize Horovod.
+hvd.init()
+```
+
+![image](https://user-images.githubusercontent.com/16640218/53518048-ff9bb780-3a84-11e9-8059-49ea1c5954dc.png)
+
+### Pin GPU to be used by each process
+
+With Horovod, you typically use a single GPU per training process:
+
+<img src="https://user-images.githubusercontent.com/16640218/53518255-7d5fc300-3a85-11e9-8bf3-5d0e8913c14f.png" width="400"></img>
+
+This allows you to greatly simplify the model, since it does not have to deal with the manual placement of tensors.  Instead, you just specify which GPU you'd like to use in the beginning of your script.
+
+Add the following code after `hvd.init()`:
+
+```python
+# Horovod: pin GPU to be used to process local rank (one GPU per process)
+config = tf.ConfigProto()
+config.gpu_options.allow_growth = True
+config.gpu_options.visible_device_list = str(hvd.local_rank())
+K.set_session(tf.Session(config=config))
+```
+
+![image](https://user-images.githubusercontent.com/16640218/53518149-4689ad00-3a85-11e9-9f59-f22eeba05e73.png)
+
+### Broadcast starting epoch from first worker to everyone else
+
+In `fashion_mnist.py`, we're using filename of the last checkpoint to determine epoch to resume training from in case of a failure:
+
+```python
+# If set > 0, will resume training from a given checkpoint.
+resume_from_epoch = 0
+for try_epoch in range(args.epochs, 0, -1):
+    if os.path.exists(args.checkpoint_format.format(epoch=try_epoch)):
+        resume_from_epoch = try_epoch
+        break
+```
+
+As you scale your workload to multi-node, some of your workers may not have access to the filesystem containing the checkpoint.  For that reason, we make the first worker to determine epoch to restart from, and *broadcast* that information to the rest of the workers.
+
+This requires adding the following code:
+
+```python
+# Horovod: broadcast resume_from_epoch from rank 0 (which will have
+# checkpoints) to other ranks.
+resume_from_epoch = hvd.broadcast(resume_from_epoch, 0, name='resume_from_epoch')
+```
+
+![image](https://user-images.githubusercontent.com/16640218/53534072-2de3bc00-3ab2-11e9-8cf1-7531542e3202.png)
+
+### Print verbose logs only on first worker
+
+Horovod uses MPI to run model training workers.  By default, MPI aggregates output from all workers.  In order to reduce clutter, it's recommended to write logs only on first worker.
```

Comment:
In order to --> To 
it's recommended to write --> we recommend that you write 
on first --> on the first